src/citrine/builders/design_spaces.py

```diff
@@ -0,0 +1,96 @@
+import pandas as pd
+from itertools import product
+
+def EnumeratedProductDesignSpaceData(dof_grids:dict):
```

Comment:
```suggestion
def enumerate_data_from_product(dof_grids:dict):
```
function names should be `snake_case` and verb-y rather than noun-y.

Reply:
What is the utility of this method over the current product design space?

Reply:
None on its own, but this one can be combined with other design spaces, and used to make Cartesian product design spaces within other functions (see the simple mixture DS)